RESTserver/src/main/java/com/softserve/if072/restservice/service/ProductService.java

```diff
@@ -20,121 +20,151 @@
         }
     }
 
     @Transactional
-    public void addProduct(Product product) {
-        productDAO.insert(product);
-    }
+    public void addProduct(Product product) { productDAO.insert(product); }
 
     @Transactional
-    public void updateProduct(Product product) throws DataNotFoundException {
-        productDAO.update(product);
+    public void updateProduct(Product product) {
+        Product getProduct = productDAO.getByID(product.getId());
+
+        if(getProduct != null) {
+            productDAO.update(product);
+            LOGGER.info(String.format("Product with ID %d was successfully updated", product.getId()));
+        } else {
+            throw new DataNotFoundException(String.format(productNotFound, product.getId()));
+        }
+
     }
 
     @Transactional
-    public void updateProductByImage(Product product) throws DataNotFoundException {
-        productDAO.updateImage(product);
+    public void updateProductByImage(Product product) {
+
+        Product getProduct = productDAO.getByID(product.getId());
+
+        if(getProduct != null) {
+            productDAO.updateImage(product);
+            LOGGER.info(String.format("Image of product with ID %d was successfully updated", product.getId()));
+        } else {
+            throw new DataNotFoundException(String.format(productNotFound, product.getId()));
+        }
+
     }
 
     @Transactional
-    public void deleteProduct(int id) throws DataNotFoundException {
+    public void deleteProduct(int id) {
+
         Product product = productDAO.getByID(id);
-        if (product != null) {
+
+        if (product != null){
             productDAO.deleteById(id);
+            LOGGER.info(String.format("Product with ID %d was successfully deleted", product.getId()));
         } else {
             throw new DataNotFoundException(String.format(productNotFound, id));
         }
     }
 
-    public List<Product> getProductsByStoreId(int storeId) throws DataNotFoundException {
+    public List<Product> getProductsByStoreId(int storeId) {
         List<Product> products = productDAO.getProductsByStoreId(storeId);
-        if (!products.isEmpty()) {
+        if (!products.isEmpty()){
             return products;
         } else {
             throw new DataNotFoundException("Products not found");
         }
     }
 
-    public List<Store> getStoresByProductId(int productId, int userId) throws DataNotFoundException {
-        List<Store> stores = productDAO.getStoresByProductIdAndUserId(productId, userId);
-        if (CollectionUtils.isNotEmpty(stores)) {
-            return stores;
+    public List<Store> getStoresByProductId(int productId, int userId) {
+
+        List<Store> storesInProduct = productDAO.getStoresByProductIdAndUserId(productId, userId);
+        List<Store> stores = new ArrayList<>();
+
+        if (!storesInProduct.isEmpty()){
+            for(Store getStore : storesInProduct) {
+                if(getStore.isEnabled()) {stores.add(getStore);}
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)